curvefs/src/metaserver/dentry_storage.cpp

```diff
@@ -133,13 +133,19 @@ void DentryList::PushBack(DentryVec* vec) {
         return;
     } else if (dentrys.size() == 0 || HasDeleteMarkFlag(*last)) {
         return;
-    } else if (onlyDir_ && last->type() != FsFileType::TYPE_DIRECTORY) {
-        return;
     } else if (last->name() == exclude_) {
         return;
     }
 
     size_++;
+
+    if (onlyDir_ && last->type() != FsFileType::TYPE_DIRECTORY) {
+        // record the last even if it is not dir(will deal in client)
```

Comment:
should you add an extra field `last` or something else in `ListDentryResponse` to indicate the start point of next list dentry instead of just adding it to current response's end?

I think it might be more clear, but you should consider whether this will introduce a compatible problem.